packages/mask/src/plugins/Avatar/Application/NFTListPage.tsx

```diff
@@ -0,0 +1,210 @@
+                      symbol: '',
+                      type: EthereumTokenType.ERC721,
+                  },
+                  {},
+                  tokenInfo.tokenId,
+              )
+            : undefined,
+    )
+
+    const onChange = (token: ERC721TokenDetailed) => {
+        if (!token) return
+        setSelectedToken(token)
+        onSelect?.(token)
+    }
+
+    const LoadStatus = Array.from({ length: 8 })
+        .fill(0)
+        .map((_, i) => (
+            <div key={i} className={classes.skeletonBox}>
+                <Skeleton animation="wave" variant="rectangular" className={classes.skeleton} />
+            </div>
+        ))
+    const Retry = (
+        <Box className={classes.error}>
+            <Typography color="textSecondary">{t.no_collectible_found()}</Typography>
+            <Button className={classes.button} variant="text" onClick={retry}>
+                {t.retry()}
+            </Button>
+        </Box>
+    )
+    return (
+        <>
+            <Box className={classes.root}>
+                <Box className={classes.gallery}>
+                    {state !== SocketState.done && collectibles.length === 0
+                        ? LoadStatus
+                        : error || collectibles.length === 0
+                        ? Retry
+                        : collectibles.map((token: ERC721TokenDetailed, i) => (
+                              <NFTImageCollectibleAvatar
+                                  key={i}
+                                  token={token}
+                                  selectedToken={selectedToken}
+                                  onChange={(token) => onChange(token)}
+                              />
+                          ))}
+                </Box>
+            </Box>
+        </>
+    )
+}
+
+export function NFTListPagePolygon(props: NFTListPageProps) {
+    const { classes } = useStyles()
+    const { address, onSelect, chainId, tokenInfo, tokens = [] } = props
+    const t = useI18N()
+    const [tokens_, setTokens_] = useState<ERC721TokenDetailed[]>(tokens)
+    const [open, setOpen] = useState(false)
+    const [selectedToken, setSelectedToken] = useState<ERC721TokenDetailed | undefined>(
+        tokenInfo
+            ? createERC721Token(
+                  {
+                      name: '',
+                      address: tokenInfo.address,
+                      chainId,
+                      symbol: '',
+                      type: EthereumTokenType.ERC721,
+                  },
+                  {},
+                  tokenInfo.tokenId,
+              )
+            : undefined,
+    )
+
+    const onChange = (token: ERC721TokenDetailed) => {
+        if (!token) return
+        setSelectedToken(token)
+        onSelect?.(token)
+    }
+
+    const onAddClick = useCallback(
+        (token: ERC721TokenDetailed) =>
+            setTokens_((tokens) => uniqBy([token, ...tokens], (x) => x.contractDetailed.address && x.tokenId)),
+        [tokens_],
```

Comment:
You add `tokens_` to the dependency array, that refresh `onAddClick` every time it invokes, since it update `tokens_` inside.

```suggestion
        [],
```